src/app/event/[eventId]/(eventId)/task/[taskId]/page.tsx

```diff
@@ -83,66 +167,158 @@ export default function TaskDetailPage({
+                                  <Button
+                                    disabled={index === 0}
+                                    onClick={handleBack}
+                                    variant="ghost"
+                                    data-testid={step.id + '-back'}
+                                  >
+                                    Back
+                                  </Button>{' '}
+                                </td>
+                              </tr>
+                            </tbody>
+                          </table>
+                        </Box>
+                      </StepContent>
+                    </Step>
+                  ))}
+                  {activeStep === steps?.length && (
+                    <Paper square elevation={0} sx={{ p: 3 }}>
+                      <Typography>
+                        All steps completed - you have finished this task
+                      </Typography>
+                    </Paper>
+                  )}
+                </Stepper>
             </div>
           )}
         </Box>
         <Link href={`/event/${eventData?.id}`}>
           <Button variant="ghost">Back to Event Page</Button>
         </Link>
+        <Box>
+          <Dialog
+            open={openPrompt}
+            onClose={handleClosePopup}
+            data-testid="login-dialog-recover"
+          >
+            <DialogTitle>Edit Step</DialogTitle>
+            <DialogActions>
+              <div className="flex flex-col gap-3" style={{ flex: 1 }}>
+                <form
+                  className="flex flex-col gap-3"
+                  onSubmit={handleSubmit(onSubmit)}
+                  data-testid="recover-account-form"
+                >
+                  <TextArea
+                    control={control}
+                    name="name"
+                    required
+                    placeholder="Enter Step Name"
+                    data-testid="name-input"
+                  />
+                  <TextArea
+                    control={control}
+                    name="description"
+                    required
+                    placeholder="Enter Step Description"
+                    data-testid="description-input"
+                  />
+                  <Button type="submit" data-testid="button-submit">
+                    Edit Step
+                  </Button>
+                </form>
+                <Button
+                  variant={'secondary'}
+                  data-testid="close-form"
+                  onClick={handleClosePopup}
+                >
+                  Close
+                </Button>
+              </div>
+            </DialogActions>
+          </Dialog>
+        </Box>
       </div>
     )
   }
 
-  return isLoading ? (
+  return (
     <div className="flex flex-col justify-center items-center min-h-[90vh]">
       <div data-testid="loader" className="loader"></div>
     </div>
-  ) : (
-    <div>{/* not found */}</div>
   )
 }
```

Comment:
Also my suggestion, you can use conditional rendering instead of using normal if (although still relevant, but this is not the best practice.

Source that will be good for consideration:
https://react.dev/learn/conditional-rendering

Also they mentioned something like this in their documentation:

> While this duplication isn’t harmful, it could make your code harder to maintain. What if you want to change the `className`? You’d have to do it in two places in your code! In such a situation, you could conditionally include a little JSX to make your code more [DRY.](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)

Reply:
No can't do, conditional rendering can't get 100% test coverage

Reply:
There will be no condition that it will get to the "Not Found" part

Reply:
And besides, even if that really goes into the "Not found" part, since there isn't any condition it will get to the loader instead

Reply:
Ok got it